frontend/internal-packages/agent/src/workflow/nodes/validateInitialSchemaNode.ts

```diff
@@ -1,14 +1,138 @@
+import { AIMessage } from '@langchain/core/messages'
 import type { RunnableConfig } from '@langchain/core/runnables'
+import { executeQuery } from '@liam-hq/pglite-server'
+import { postgresqlSchemaDeparser, schemaSchema } from '@liam-hq/schema'
+import * as v from 'valibot'
+import { getConfigurable } from '../../chat/workflow/shared/getConfigurable'
 import type { WorkflowState } from '../../chat/workflow/types'
+import { WorkflowTerminationError } from '../../shared/errorHandling'
+
+const isFirstWorkflowExecution = (state: WorkflowState): boolean => {
+  return !state.messages.some((msg) => msg instanceof AIMessage)
+}
 
 /**
  * Validates initial schema and provides Instant Database initialization experience.
  * Only runs on first workflow execution.
  */
 export async function validateInitialSchemaNode(
   state: WorkflowState,
-  _config: RunnableConfig,
+  config: RunnableConfig,
 ): Promise<WorkflowState> {
-  // TODO: Implement validation logic
+  if (!isFirstWorkflowExecution(state)) {
+    return state
+  }
```

Comment:
This check looked simpler to express as conditionEdge. What do you think?

```mermaid
flowchart TD
    START --> isFirstExecution{isFirstExecution?}
    isFirstExecution -->|Yes| validateInitialSchema[validateInitialSchema]
    isFirstExecution -->|No| classifyMessage[classifyMessage]
    validateInitialSchema --> classifyMessage
```

Reply:
You were absolutely right. I hadn't considered that point. Thank you for pointing it out!
https://github.com/liam-hq/liam/pull/3331/commits/7d6c7ab4c73efeb7b586e360e05da25f198dfe6f